docs/ConcurrencyRoadmap.md

```diff
@@ -0,0 +1,265 @@
+
+The problem of actor isolation reduces to the problem of ensuring that all ordinary mutable memory is accessed only by a particular actor or task.  This in turn reduces to an analysis of how the memory is accessed and who can access it in the first place. We can categorize memory into a few groups:
+
+* Properties of an actor will be protected by the actor.
+* Immutable memory (such as a `let` constant), local memory (such as a local variable that’s never captured), and value component memory (such as a properties of a struct, or an enum case), are already protected from data races.
+* Unsafe memory (such as an arbitrary allocation referenced by an `UnsafeMutablePointer`) is associated with an unsafe abstraction.  It’s actively undesirable to try to force these abstractions to be used safely, because these abstractions are meant to be usable to bypass safe language rules when necessary. Instead, we have to trust the programmer to use these correctly.
+* Global memory (such as a global or static variable) can in principle be accessed by any code anywhere, so is subject to data races.
+* Class component memory can also be accessed from any code that hold a reference to the class. This means that while the _reference_ to the class may be protected by an actor, passing that reference between actors exposes its properties to data races. Note that this also references to classes held within value types when these are passsed between actors.
+
+The goal of **full actor isolation** is to ensure that these last two categorizations are protected by default.
+
+### First Phase: Basic Actor Isolation
+
+The first phase introduces safety benefits. Users will be able to protect  global variables with global actors, and to protect class members by converting them to actor classes. Frameworks that require access on a particular queue can define a global actor and default protocols to it.
+
+A number of important cases of actor isolation will be enforced in this stage:
+
+```swift
+actor class MyActor {
+  let immutable: String = "42"
+  var mutableArray: [String] = []
+
+  func synchronousFunction() {
+	  mutableArray += ["syncFunction called"]
+  }
+}
+
+extension MyActor {
+
+  func asyncFunction(otherActor: MyActor) async {
+    // allowed: an actor can access its internal state, even in an extension
+    self.mutableArray += ["asyncFunction called"]
+  
+    // allowed: immutable memory can be accessed from outside the actor
+    print(other.immutable)
+
+    // error: an actor cannot access another's mutable state
+    otherActor.mutableArray += ["not allowed"]
+
+    // error: either reading or writing
+    print(other.mutableArray.first)
+    
+    // allowed: async functions can call async functions on other actors
+    other.asyncFunction(otherActor: self)
+    
+    // error: only asynchronous functions can be called from outside the actor
```

Comment:
> only asynchronous functions can be called **from outside** the actor

only asynchronous functions can be called **inside** the actor

Reply:
No, the phrasing is correct here.

This is attempting to call a synchronous function on the `other` actor -- this we are outside of it.